lib/barbeque/message_queue.rb

```diff
@@ -23,9 +25,22 @@ def dequeue
         valid_messages.concat(messages)
       end
 
-      valid_messages.shift.tap do |message|
-        delete_message(message)
+      message = valid_messages.shift
+      # XXX: #receive_messages returns one message at most because it calls
+      #       receive_message API without max_number_of_messages option.
+      unless valid_messages.empty?
+        extend_visibility_timeout(valid_messages)
```

Comment:
When ReceiveMessage returns several messages, Barbeque::MessageQueue#dequeue returns one message and holds other messages in memory. While the returned message has reasonable visibility timeout (SQS queue's visibility timeout), the held messages don't because they are processed in the next barbeque-worker loop.
I think visibility timeout of the held messages should be extended in each loop.
(However, I noticed that `valid_messages.size` never becomes more than one  😇 )